### DISCUSSION TITLE
Mozilla Firefox web browser occasionally crashes and is unstable in Kali Linux x64 VM in UTM Emulation

### DISCUSSION THREAD

**Discussion by MinhPham123456789** [utmapp/UTM#5868] [Linux]
**Describe the issue**  
Mozilla Firefox web browser is not stable and occasionally crashes in Kali Linux x64 architecture VM in UTM using emulation.
In the advanced option there are many cpu types and cpu flags, is there any combination to support the speed and stablize the VM?
Note: the VM is running normally and is stable, just when it runs firefox web browser, the web browser is unstable.

**Configuration**  
* UTM Version: 4.3.5
* macOS Version: 14.0 Sonoma
* Mac Chip: M2

**Crash log**  
VM does not crash, just the program inside the VM.

**VM configuration**
* Architecture: x86_64
* System: Standard PC (Q35 + ICH9, 2009) (alias of pc-q35-7.2) (q35)
* RAM: 8GB, Core: 4
* Display: virtio-vga-gl (GPU Supported)
* CPU: QEMU Virtual CPU version 2.5+ (qemu64-v1)
* Force Multicore
* No cpu flag selected

**Comment by osy** [comment]
Emulation in general is not recommended for CPU intensive tasks (like Firefox). You are probably better off using an older web browser.
